pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -288,6 +298,7 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
             sv.updateData(data_storage.data(), data_storage.size());
             obs.applyInPlaceShots(sv, eigenvalues, obs_wires);
             Derived measure(sv);
+            measure.setSeed(this->_deviceseed);
```

Comment:
Isn't it circular here? i.e. the `setSeed` method will record the received seed into `this->_deviceseed`, but here the setter argument is also `this->_deviceseed`. I know in the catalyst path the measurement object's `_deviceseed` is set from the catalyst rng's produced number, but what sets the seed for the non-catalyst path? Because here it seems like the seed will always be the initialized value of `_deviceseed` when `measure` is created (which is nullopt).

Reply:
`this->_deviceseed` contains the seed given by the simulator. When we create the new Derived measure here, the seed is `nullopt `for that new measure. Hence, we need to set it with the current value of the seed. If we don't do this here, `expval`, `var` and `probs` will be unseeded.

Reply:
Ahh, the link I was missing was that the `Derived measure` is a separately created object whose seed need to be set again. Thanks!